### ISSUE TITLE
Symbol/WordArray style "percent" should also consider [ ] chars as "bad"

### ISSUE THREAD

**Issue by johnnyshields** [rubocop/rubocop#11802]
The current behavior of the Symbol/WordArray cop's default style ("percent") is as follows:

```ruby
# bad
[:foo, :[]=]

# bad (contains spaces)
%w[foo\ bar baz\ quux]

# good (allows [ ] chars)
%i[foo \[\]=]
```

I think we should change this so that `[ ]` trigger the same special handling that space char does, and the preferred style here would be `[:foo, :[]=]`

**Comment by bbatsov** [comment]
Makes sense to me. PR welcome!

**Comment by johnnyshields** [comment]
@koic Awesome, thank you!
